src/Network/Probe.php

```diff
@@ -58,26 +59,23 @@ class Probe
 	/**
 	 * Remove stuff from an URI that doesn't belong there
 	 *
-	 * @param string $URI
+	 * @param string $uri
 	 * @return string Cleaned URI
 	 */
-	public static function cleanURI(string $URI)
+	public static function cleanURI(string $uri): string
 	{
 		// At first remove leading and trailing junk
-		$URI = trim($URI, "@#?:/ \t\n\r\0\x0B");
-
-		$parts = parse_url($URI);
+		$uri = trim($uri, "@#?:/ \t\n\r\0\x0B");
 
-		if (empty($parts['scheme'])) {
-			return $URI;
+		$uri = new Uri($uri);
```

Comment:
What is the advantage of not using `parse_url` here?

Reply:
Anytime we need to actually change individual parts of a URI, it's better to use this object than `Network::unparseURL` thanks to compliance checks we do not currently perform.

For simple read access, it's equivalent, which is why I don't particularly recommend to change the places we already use `parse_url`.

Reply:
BTW: It's slightly confusing when the `$uri` string is converted to a `$uri` object.

Reply:
I'm not happy with it either but I couldn't come with a good way to separate URI strings and URI objects.

Reply:
Couldn't you use some prefix for the parameter text? I find it really confusing.

Reply:
I don't like prefixes and I use suffixes sparingly. Since we do not have any further use for the URI string after it's converted to an object, I don't think it is a problem in this short method.

Reply:
A good alternative is for the method to take a `Uri` object as parameter, it would require changing only 6 mentions, what do you say?

Reply:
Wouldn't `$rawuri` be a good name?

Reply:
I guess so, let's go with that.